File: Pump Fun/typescript/examples/lock.ts
import { JellychainPumpFunSDK } from '../src';

async function main() {
  const sdk = new JellychainPumpFunSDK({
    rpcUrl: process.env.SOLANA_RPC_URL || 'https://api.mainnet-beta.solana.com',
  });

  const wallet = sdk.walletFromBase58(process.env.PRIVATE_KEY_BASE58!);
  const mintAddress = process.env.MINT_ADDRESS!;

  console.log('[JellyChain] Locking dev tokens...');
  console.log('Wallet:', wallet.publicKey.toBase58());
  console.log('Mint:', mintAddress);

  const result = await sdk.lock({
    wallet: wallet.payer,
    mintAddress,
    lockType: 'burn',
  });

  if (result.success) {
    console.log('Tokens locked!');
    console.log('Locked amount:', result.lockedAmount);
    console.log('Lock type:', result.lockType);
    console.log('TX:', result.txSignature);
    console.log('Solscan:', result.solscanUrl);
  } else {
    console.error('Lock failed:', result.error);
  }
}

main().catch(console.error);

File: Pump Fun/python/examples/lock.py
"""
Example: Lock (burn) dev tokens on Pump.fun using JellyChain Python SDK

Docs:     https://jellychain.fun/docs
Terminal: https://terminal.jellychain.fun
GitHub:   https://github.com/jelly-chain
"""
import os
from jellychain_pumpfun import JellychainPumpFunSDK, LockParams


def main() -> None:
    sdk = JellychainPumpFunSDK(
        rpc_url=os.getenv("SOLANA_RPC_URL", "https://api.mainnet-beta.solana.com"),
    )

    private_key = os.environ["PRIVATE_KEY_BASE58"]
    mint_address = os.environ["MINT_ADDRESS"]

    wallet = sdk.wallet_from_base58(private_key)
    print(f"[JellyChain] Locking dev tokens...")
    print(f"  Wallet: {wallet.public_key}")
    print(f"  Mint:   {mint_address}")

    result = sdk.lock(LockParams(
        private_key_base58=private_key,
        mint_address=mint_address,
        lock_type="burn",
    ))

    if result.success:
        print("Tokens locked!")
        print(f"  Locked amount: {result.locked_amount}")
        print(f"  Lock type:     {result.lock_type}")
        print(f"  TX:            {result.tx_signature}")
        print(f"  Solscan:       {result.solscan_url}")
    else:
        print(f"Lock failed: {result.error}")


if __name__ == "__main__":
    main()
